### DISCUSSION TITLE
oracle cdc 如何能拿到变更操作的发生时间

### DISCUSSION THREAD

**Discussion by MonsterChenzhuo** [apache/flink-cdc#1214] [Q&A]
我通过dubug调试进入到
<img width="1180" alt="图片" src="https://user-images.githubusercontent.com/60029759/170027849-998c74bb-bac3-4ffe-a576-044a08cf9b43.png">

SourceRecord{sourcePartition={server=oracle_logminer}, sourceOffset={snapshot=true, scn=6338831463326, snapshot_completed=false}} ConnectRecord{topic='oracle_logminer.xxxxx', kafkaPartition=null, key=Struct{ID=529}, keySchema=Schema{oracle_logminer.xxxxx.Key:STRUCT}, value=Struct{after=Struct{xxxx},source=Struct{version=1.5.4.Final,connector=oracle,name=oracle_logminer,ts_ms=1653392220143,snapshot=true,db=SVDP,schema=HLASSET,table=BZ_HOUSE,scn=6338831463326},op=r,ts_ms=1653392220149}, valueSchema=Schema{oracle_logminer.xxxxx.Envelope:STRUCT}, timestamp=null, headers=ConnectHeaders(headers=)}

发现ts_ms 是当前时间，我现在想获取到这个R操作的发生时间，求教该如何获取？

**Comment by shadon178** [comment]
R操作是全量阶段，并不是从binlog里面获取到的，而是select查询出来的数据。这个查询时间怕是不好获取。

**Reply by MonsterChenzhuo** [reply]
@shadon178 
<img width="1296" alt="图片" src="https://user-images.githubusercontent.com/60029759/170182154-987e71c6-bfaf-408b-935f-f00e0d436366.png">
您好我理解sourcets_ms{}中的ts_ms是操作记录变更的时间，而第二个ts_ms是cdc处理这条记录的时间，不知道是否正确

**Reply by MonsterChenzhuo** [reply]
感谢您的解答，我这才理解全量阶段的原理，现在我仅需要的是增量阶段的操作时间

**Comment by MonsterChenzhuo** [comment]
@shadon178 您好
2022-05-25 12:50:37,439 INFO  io.debezium.connector.oracle.logminer.LogMinerQueryResultProcessor [] - DDL: transactionId=c4011c00f4280000, SCN=6338900744232, table_name=LOGMNRLT_179_DICTIONARY$, segOwner=SYS, operationCode=5, offsetSCN=6338900724993,  commitOffsetSCN=6338900721105, REDO_SQL: CREATE TABLE "SYS"."LOGMNRLT_179_DICTIONARY$" TABLESPACE "SYSAUX" LOGGING AS SELECT * FROM "SYSTEM"."LOGMNR_DICTIONARY$" WHERE 1 = 2;
我看程序一直打印这个日志，您清楚这个是什么逻辑
